mlflow/johnsnowlabs/__init__.py

```diff
@@ -837,14 +874,21 @@ def __init__(
 
     def predict(self, text, params: Optional[Dict[str, Any]] = None):
         """
-        Generate predictions given input data in a pandas DataFrame.
-
-        :param text: pandas DataFrame containing input data.
-        :param params: Additional parameters to pass to the model for inference.
-
-                       .. Note:: Experimental: This parameter may change or be removed in a future
-                                               release without warning.
-        :return: List with model predictions.
+             Generate predictions given input data in a pandas DataFrame.
+             1) If df contains any column that is in PredictParams fields,
+              the first row will be parsed as parameters
+             2) If df contains column `file` and `file_type`
+              columns, each row will be deserialized into file
+             :param text: pandas DataFrame containing input data.
+             :param params: Additional parameters to pass to the model for inference.
+
+        .. Note:: Experimental: This parameter may change or be removed in a future
+                                release without warning.
+             :return: List with model predictions.
         """
-        output_level = params.get("output_level", "") if params else ""
-        return self.spark_model.predict(text, output_level=output_level).reset_index().to_json()
+        params = PredictParams.maybe_from_pandas_df(text)
```

Comment:
Instead of converting params from pandas dataframe, could we directly utilize the `params` variable as you're passing a dictionary anyways. You could check the allowable params inside `predict` function or you could set a default ModelSignature with ParamsSchema (example here https://mlflow.org/docs/latest/models.html#id7) in johnsnowlabs flavor and we automatically validate the params against defined types.

Reply:
Thanks makes sense let me look into the signatures

Reply:
Added some changes for signatures, still need to test tough coming back with results soon